crates/swc_ecma_transforms_compat/src/es2020/opt_chaining.rs

```diff
@@ -501,16 +503,24 @@ impl OptChaining {
                 ..
             }) => {
                 let obj_span = obj.span();
-                let is_super_access = match **obj {
+                let is_super_access = match obj.as_ref() {
                     Expr::Member(MemberExpr {
                         obj: ExprOrSuper::Super(..),
                         ..
                     }) => true,
                     _ => false,
                 };
 
-                let (left, right, alt) = match **obj {
-                    Expr::Ident(..) => (obj.clone(), obj.clone(), e.expr.take()),
+                let (left, right, alt) = match obj.as_ref() {
+                    Expr::Ident(ident) => (
+                        obj.clone(),
+                        obj.clone(),
+                        if *ident.sym == *"eval" {
```

Comment:
This comparison can be replaced with `ident.sym == js_word!("eval")` which is much faster.